ocs_ci/ocs/amq.py

```diff
@@ -446,7 +447,170 @@ def run_in_bg(self, namespace=constants.AMQ_NAMESPACE, value='10000', since_time
+            test_duration_minutes (int): Time to run the workloads
+
+        Return:
+            result (json): Returns the test result
+
+        """
+
+        # Namespace for to helm/tiller
+        try:
+            self.create_namespace(tiller_namespace)
+        except CommandFailed as ef:
+            if f'project.project.openshift.io "{tiller_namespace}" already exists' not in str(ef):
+                raise ef
+
+        # Create rbac file
+        try:
+            sa_tiller = list(templating.load_yaml(constants.AMQ_RBAC_YAML, multi_document=True))
+            sa_tiller[0]["metadata"]["namespace"] = tiller_namespace
+            sa_tiller[1]["subjects"][0]["namespace"] = tiller_namespace
+            self.sa_tiller = OCS(**sa_tiller[0])
+            self.crb_tiller = OCS(**sa_tiller[1])
+            self.sa_tiller.create()
+            self.crb_tiller.create()
+        except(CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during creation of service account tiller')
+            raise cf
+
+        # Install helm cli (version v2.16.0 as we need tiller component)
+        # And create tiller pods
+        url = "https://get.helm.sh/helm-v2.16.1-linux-amd64.tar.gz"
+        try:
+            log.info(f'Install helm on {self.dir}')
+            wget_cmd = f"wget -c --read-timeout=5 --tries=0 {url}"
+            untar_cmd = "tar -zxvf helm-v2.16.1-linux-amd64.tar.gz"
+            tiller_cmd = (
+                f"linux-amd64/helm init --tiller-namespace {tiller_namespace}"
+                f" --service-account {tiller_namespace}"
+            )
+            run(
+                wget_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+            run(
+                untar_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+            run(
+                tiller_cmd,
+                shell=True,
+                cwd=self.dir,
+                check=True
+            )
+        except (CommandFailed, CalledProcessError)as cf:
+            log.error('Error during installing helm')
+            raise cf
+
+        # Validate tiller pod is running
+        log.info("Waiting for 30s")
+        time.sleep(30)
+        if self.is_amq_pod_running(pod_pattern="tiller", expected_pods=1, namespace=tiller_namespace):
+            log.info("Tiller pod is running")
+        else:
+            raise ResourceWrongStatusException("Tiller pod is not in running state")
+
+        # Create benchmark pods
+        log.info("Create benchmark pods")
+        try:
+            values = templating.load_yaml(constants.AMQ_BENCHMARK_VALUE_YAML)
+            values["numWorkers"] = num_of_clients
+            benchmark_cmd = (
+                f"linux-amd64/helm install {constants.AMQ_BENCHMARK_POD_YAML}"
+                f" --name {name} --tiller-namespace {tiller_namespace}"
+            )
+            run(
```

Comment:
Any reason why you are not using exec_cmds which are already present in ocs-ci?
This comment is for all the places where you are using run

Reply:
Changed to exec_cmds, so marking as resolved